frontend/src/components/customer/PaymentMethodsCard/PaymentMethodsCard.tsx

```diff
@@ -0,0 +1,353 @@
+        const newChecked = [...checked];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setChecked(newChecked);
+    };
+
+    const handleSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+        setFormErrors({});
+        const errors: Record<string, string> = {};
+
+        if (!newPayment.card_number) {
+            errors.card_number = "Card No. is required";
+        } else if (newPayment.card_number.length !== 16) {
+            errors.card_number = "Card No. must be 16 digits";
+        } else if (!/^[0-9]+$/.test(newPayment.card_number)) {
+            errors.card_number = "Invalid card number";
+        }
+
+        if (!newPayment.cvv) {
+            errors.cvv = "CVV is required";
+        } else if (newPayment.cvv.length !== 3) {
+            errors.cvv = "CVV must be three digits";
+        } else if (!/^[0-9]+$/.test(newPayment.cvv)) {
+            errors.cvv = "Invalid CVV";
+        }
+        if (!newPayment.card_name) {
+            errors.card_name = "Card Name is required";
+        }
+
+        if (!newPayment.expiration_date) {
+            errors.expiration_date = "Expiration Date is required";
+        } else if (newPayment.expiration_date.length !== 7) {
+            errors.expiration_date = "Invalid Expiration Date";
+        } else if (newPayment.expiration_date.charAt(2) !== "/") {
+            errors.expiration_date = "Invalid Expiration Date";
+        } else {
+            const [month, year] = newPayment.expiration_date.split("/");
+            const inputDate = new Date(Number(year), Number(month) - 1);
+            const currentDate = new Date(
+                new Date().getFullYear(),
+                new Date().getMonth()
+            );
+            if (inputDate < currentDate) {
+                errors.expiration_date = "Invalid Expiration Date";
+            }
+        }
+
+        if (Object.keys(errors).length > 0) {
+            setFormErrors(errors);
+        } else {
+            console.log("New Payment Data:", newPayment);
+
+            const updatedPaymentsData = [...paymentsData, newPayment];
+            setPaymentsData(updatedPaymentsData);
+
+            setNewPayment({
+                card_number: "",
+                card_name: "",
+                cvv: "",
+                expiration_date: "",
+            });
+        }
+    };
+    const handleCancel = () => {
+        setNewPayment({
+            card_number: "",
+            card_name: "",
+            cvv: "",
+            expiration_date: "",
+        });
+    };
+    return (
+        <form onSubmit={handleSubmit}>
+            <Paper elevation={2} sx={{ mt: "3rem", mb: "3rem" }}>
+                <Grid container direction="column" gap={2}>
+                    <Grid container direction="column">
+                        <Grid item sx={{ ml: "2rem", mr: "2rem", mt: "2rem" }}>
```

Comment:
Same problem, make use and `mx` and `my`. You can even use `margin:"2rem 2rem 2rem 0"` if I am not wrong